src/macaron/slsa_analyzer/git_url.py

```diff
@@ -106,76 +107,59 @@ def check_out_repo_target(git_obj: Git, branch_name: str = "", digest: str = "",
             return False
 
-    # Checkout the resolved branch from remote origin.
-    # The name of each remote ref will be in the form <prefix>/<name> (e.g. origin/master).
-    # We only work with origin remote in Macaron.
-    try:
-        org_remote: Remote = git_obj.repo.remote("origin")
-    except ValueError as error:
-        logger.error("Cannot get origin remote from the repository.")
-        logger.error(error)
-        return False
-
-    if not org_remote or not hasattr(org_remote, "refs"):
-        logger.critical("The origin remote does not exist or does not have the references attribute.")
-        return False
-
-    if res_branch in [os.path.basename(ref.name) for ref in org_remote.refs]:
-        logger.info(
-            "Find branch %s in the remote's refs of the local repository. The branch is checked out directly.",
-            res_branch,
-        )
-        git_obj.repo.git.checkout(res_branch)
-    else:
-        # This might happen because the git repository is a shallow clone.
-        logger.info("Cannot find branch %s. Try to fetch the branch from remote.", res_branch)
+    if not offline_mode:
+        # Fetch from remote by running ``git fetch`` inside the target repository.
+        # We don't specify any remote name (e.g. origin) is because we want git to resolve the default fetching
+        # target by itself.
+        # For example, the user runs Macaron on a local repository where the remote is set to have name "foo_origin"
+        # instead.
+        # References: https://git-scm.com/docs/git-fetch
         try:
-            # Trying to get this branch by fetching from remote and checkout the branch again.
-            # If we can't checkout, we consider that this branch does not exist.
-            # https://stackoverflow.com/questions/23708231/git-shallow-clone-clone-depth-misses-remote-branches.
-            git_obj.repo.git.remote("set-branches", "--add", "origin", res_branch)
-            org_remote.fetch(res_branch)
-            git_obj.repo.git.checkout(res_branch)
+            git_obj.repo.git.fetch()
         except GitCommandError as error:
-            logger.error("Error while getting branch %s. Error: %s", res_branch, error)
-            return False
-
-    logger.info("Successfully checked out branch %s.", res_branch)
-
-    # We only pull the latest changes if we are not running in offline mode and:
-    #   - no digest is provided.
-    #   - or a commit digest is provided but it does not exist in the current local branch.
-    if not offline_mode and (not digest or not commit_exists(git_obj, digest)):
-        logger.info("Pulling the latest changes of branch %s fast-forward only.", res_branch)
-        if not pull_latest_changes(git_obj):
-            logger.error(
-                "Cannot pull the latest changes for branch %s.",
-                res_branch,
-            )
+            logger.error("Cannot perform fetching on this repository. Error: %s", error)
```

Comment:
```suggestion
           logger.error("Unable to fetch from the remote repository. Error: %s", error)
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/491/commits/7b35e849e44cc4bdfd4b23499db6c16430b55270